File: frontend/src/types/staff-service.ts
export interface StaffServiceAssignment {
  id: string
  staff_id: string
  service_id: string
  staff_name: string
  service_name: string
  created_at: string
  updated_at: string
}

export interface AssignServiceToStaffRequest {
  service_id: string
}

export interface AssignMultipleServicesToStaffRequest {
  service_ids: string[]
}

export interface StaffWithServices {
  id: string
  business_id: string
  first_name: string
  last_name: string
  full_name: string
  phone?: string
  gender?: 'male' | 'female' | 'other' | 'prefer_not_to_say'
  position: string
  description?: string
  specialization?: string
  is_active: boolean
  services: Array<{
    id: string
    business_id: string
    name: string
    duration_min: number
    price_cents: number
    created_at: string
    updated_at: string
  }>
  created_at: string
  updated_at: string
}

File: internal/dto/staff_service.go
package dto

// AssignServiceToStaffRequest represents the request to assign a service to staff
type AssignServiceToStaffRequest struct {
	ServiceID string `json:"service_id" validate:"required,uuid4"`
}

// AssignMultipleServicesToStaffRequest represents the request to assign multiple services to staff
type AssignMultipleServicesToStaffRequest struct {
	ServiceIDs []string `json:"service_ids" validate:"required,dive,uuid4"`
}

// StaffServiceResponse represents a staff-service relationship in API responses
type StaffServiceResponse struct {
	ID          string `json:"id"`
	StaffID     string `json:"staff_id"`
	ServiceID   string `json:"service_id"`
	StaffName   string `json:"staff_name"`
	ServiceName string `json:"service_name"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

// StaffWithServicesResponse represents a staff member with their assigned services
type StaffWithServicesResponse struct {
	ID             string            `json:"id"`
	BusinessID     string            `json:"business_id"`
	FirstName      string            `json:"first_name"`
	LastName       string            `json:"last_name"`
	FullName       string            `json:"full_name"`
	Phone          string            `json:"phone,omitempty"`
	Gender         string            `json:"gender,omitempty"`
	Position       string            `json:"position"`
	Description    string            `json:"description,omitempty"`
	Specialization string            `json:"specialization,omitempty"`
	IsActive       bool              `json:"is_active"`
	Services       []ServiceResponse `json:"services"`
	CreatedAt      string            `json:"created_at"`
	UpdatedAt      string            `json:"updated_at"`
}
